pkg/frontend/ready_get.go

```diff
@@ -16,3 +16,11 @@ func (f *frontend) getReady(w http.ResponseWriter, r *http.Request) {
 		api.WriteError(w, http.StatusInternalServerError, api.CloudErrorCodeInternalServerError, "", "Internal server error.")
 	}
 }
+
+func (f *frontend) getLive(w http.ResponseWriter, r *http.Request) {
+	if f.backendLive() {
```

Comment:
Currently frontend and backend do not directly depend on each other: they communicate via DB. Despite the fact that they share common packages/states (like `env`), I think it is currently relatively easy to split them into two binaries. One of the benefits I see here is ability to deploy backend separately from frontend and, for example, have 3 VMs running frontend and 5 VMs running backend. If I understand correctly, with this change frontend will be reading memory written by backend. I'm worried that with more dependencies between frontend and backend we will have hard times scaling horizontally. What do you think? Should we be concerned about scaling at this stage or not?

Another more high level question: why do we have a need for introducing a logical dependency? I think frontend still should be able to accept requests, perform validation, create\update objects in the DB while backend is not ready/live. Is it becasue we want to avoid the situation when users build a queue of things for backends to handle while backend is down?

Reply:
This is needed so we would know if we can stop RP VM running both binaries. An alternative to this is to call `RunCommand` VMSS module and check if pid is active. Or `systemd` to power off the VM when RP is `SIGTERM`ed, but this does not look right to me too as it is introducing one more layer of complexity. This is a nasty solution, agreed. 

Situation: We want to bring new RP backend into the mix. When a new one is ready, we need to terminate gradually old ones. We do this by sending `SIGTERM` command. The first frontend stops advertising `ready` endpoint so no new requests will come. Second `Backend` is moved to `stopping` stage and will not pick any new work. But some work might be still in-progress. 
And there is no easy way to know from outside if backend already finished or not as at this point FE is not running anymore. 

I think if we end-up splitting those 2 processes, this can be replaced by either `gRPC` call or backend will have its own small webserver running and advertising health. 

If you have better idea how to solve this - let me know :/

Reply:
Alternative could be to check if anything owns a document in DB based on backend ID. But this would require setting ID for backend on the initiation. And if we split binaries, it is still a throw-away code ;/

Reply:
redid